packages/cosmic-swingset/tools/test-kit.js

```diff
@@ -0,0 +1,388 @@
+/* eslint-env node */
+
+import * as fsPromises from 'node:fs/promises';
+import * as pathNamespace from 'node:path';
+import { assert, Fail } from '@endo/errors';
+import * as ActionType from '@agoric/internal/src/action-types.js';
+import { makeBootMsg } from '@agoric/internal/src/chain-utils.js';
+import { initSwingStore } from '@agoric/swing-store';
+import { makeSlogSender } from '@agoric/telemetry';
+import { launch } from '../src/launch-chain.js';
+import { DEFAULT_SIM_SWINGSET_PARAMS } from '../src/sim-params.js';
+import { makeBufferedStorage } from '../src/helpers/bufferedStorage.js';
+import { makeQueue, makeQueueStorageMock } from '../src/helpers/make-queue.js';
+
+/** @import { BlockInfo, BootMsg } from '@agoric/internal/src/chain-utils.js' */
+/** @import { SwingSetConfig } from '@agoric/swingset-vat' */
+
+/**
+ * @template T
+ * @typedef {(input: T) => T} Replacer
+ */
+
+/** @type {Replacer<object>} */
+const clone = obj => JSON.parse(JSON.stringify(obj));
```

Comment:
Right, because it's only used on `defaultBootMsg` and `baseConfig`, both defined in this file (and the latter of the two must be JSONable by definition).